drivers/bluetooth/hci/userchan.c

```diff
@@ -58,15 +58,51 @@ static unsigned int port;
 static char socket_path[UNIX_ADDR_BUFF_SIZE];
 static bool arg_found;
 
+
+static bool is_hci_event_discardable(const uint8_t *evt_data)
+{
+	uint8_t evt_type = evt_data[0];
+
+	switch (evt_type) {
+#if defined(CONFIG_BT_CLASSIC)
+	case BT_HCI_EVT_INQUIRY_RESULT_WITH_RSSI:
+	case BT_HCI_EVT_EXTENDED_INQUIRY_RESULT:
+		return true;
+#endif
+	case BT_HCI_EVT_LE_META_EVENT: {
+			uint8_t subevt_type = evt_data[sizeof(struct bt_hci_evt_hdr)];
+
+		switch (subevt_type) {
+		case BT_HCI_EVT_LE_ADVERTISING_REPORT:
+			return true;
+#if defined(CONFIG_BT_EXT_ADV)
+		case BT_HCI_EVT_LE_EXT_ADVERTISING_REPORT:
+		{
+			const struct bt_hci_evt_le_ext_advertising_report *ext_adv =
+					(void *)&evt_data[3];
+
+			return (ext_adv->num_reports == 1) &&
+					((ext_adv->adv_info[0].evt_type &
+							BT_HCI_LE_ADV_EVT_TYPE_LEGACY) != 0);
```

Comment:
Something's not quite right with the alignment here. The convention is to align with the appropriate opening parenthesis using N tabs + 0-7 spaces. You might be able to make your editor do the right thing by putting the entire value behind a top-level parenthesis. That way e.g. the second line wont start at an arbitrary point.

Reply:
this is just cnp from IPC :)

I ran clang-format and I'm not sure I like what it did with that part : 

```
			return (ext_adv->num_reports == 1) &&
			       ((ext_adv->adv_info[0].evt_type & BT_HCI_LE_ADV_EVT_TYPE_LEGACY) !=
				0);
```